### PR TITLE
adds distinct plugin config and factory with path

### PR DESCRIPTION
**Pull request by jpfaria**

### CODE REVIEW

File: google.golang.org/grpc.v1/server/plugins/contrib/opentracing-contrib/go-grpc.v0/opentracing.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+func NewNewrelicWithConfigPath(path string) (*Newrelic, error) {
+	o, err := NewOptionsWithPath(path)
+	if err != nil {
+		return nil, err
+	}
+	return NewNewrelicWithOptions(o), nil
+}
+
+func NewNewrelic() *Newrelic {
```
Comment (jvitoroc) [inline google.golang.org/grpc.v1/server/plugins/contrib/opentracing-contrib/go-grpc.v0/opentracing.go:37]: newrelic? acho que o nome do plugin seria outro

File: google.golang.org/grpc.v1/client/plugins/contrib/newrelic/go-agent.v3/newrelic.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	h := NewDatadogWithOptions(o)
+	return h.Register(ctx)
+}
+
+type Datadog struct {
+	options *Options
+}
+
+func NewDatadogWithOptions(options *Options) *Datadog {
```
Comment (jvitoroc) [inline google.golang.org/grpc.v1/client/plugins/contrib/newrelic/go-agent.v3/newrelic.go:25]: É newrelic o nome do plugin

File: go-chi/chi.v5/plugins/contrib/americanas-go/rest-response.v1/rest-response.go [hunk 3]
```diff
@@ -12,37 +12,58 @@ import (
...
-	if !IsEnabled() {
+	l := NewStatus()
+	return l.Register(ctx)
+}
+
+type Status struct {
+	options *Options
+}
+
+func NewStatusWithOptions(options *Options) *Status {
```
Comment (jvitoroc) [inline go-chi/chi.v5/plugins/contrib/americanas-go/rest-response.v1/rest-response.go:23]: O nome do plugin é status mesmo?

File: go-redis/redis.v7/plugins/contrib/americanas-go/health.v1/client_health.go [hunk 4]
```diff
@@ -39,3 +47,12 @@ func (i *ClientHealth) Register(ctx context.Context, client *redis.Client) error
 
 	return nil
 }
+
+func ClientRegister(ctx context.Context, client *redis.Client) error {
```
[the remark this answers is not in the document]
Comment (jpfaria) [reply go-redis/redis.v7/plugins/contrib/americanas-go/health.v1/client_health.go:51]: it cannot have two functions with the same name.

File: go-redis/redis.v7/plugins/contrib/americanas-go/health.v1/cluster_health.go [hunk 5]
```diff
@@ -39,3 +47,12 @@ func (i *ClusterHealth) Register(ctx context.Context, client *redis.ClusterClien
 
 	return nil
 }
+
+func ClusterRegister(ctx context.Context, client *redis.ClusterClient) error {
```
[the remark this answers is not in the document]
Comment (jpfaria) [reply go-redis/redis.v7/plugins/contrib/americanas-go/health.v1/cluster_health.go:51]: it cannot have two functions with the same name.

File: google.golang.org/grpc.v1/server/plugins/contrib/opentracing-contrib/go-grpc.v0/opentracing.go (hunk 1 above)
Comment (jpfaria) [reply google.golang.org/grpc.v1/server/plugins/contrib/opentracing-contrib/go-grpc.v0/opentracing.go:37]: good! :)

File: google.golang.org/grpc.v1/client/plugins/contrib/newrelic/go-agent.v3/newrelic.go (hunk 2 above)
Comment (jpfaria) [reply google.golang.org/grpc.v1/client/plugins/contrib/newrelic/go-agent.v3/newrelic.go:25]: good

File: go-chi/chi.v5/plugins/contrib/americanas-go/rest-response.v1/rest-response.go (hunk 3 above)
Comment (jpfaria) [reply go-chi/chi.v5/plugins/contrib/americanas-go/rest-response.v1/rest-response.go:23]: yeap!!

**Pull request closed without merging**